lib/src/authenticate.tsx

```diff
@@ -103,6 +103,43 @@ const AuthProvider: FunctionComponent<PropsWithChildren<AuthProviderPropsInterfa
         setConfigState(config);
     }, [ config ]);
 
+    /**
+     * Subscribe to the URL search param changes and try signing in whenever required.
+     */
+    useEffect(() => {
```

Comment:
I am not sure if this would create the subscription pattern since the hook is going to be called only when this component is rendered.  And `SPAUtils.hasAuthSearchParamsInURL()` is not going to trigger a rerender.

Reply:
Removed this since it doesn't catch history push events. And we don't have a usecase yet for it.